pkg/util/validation/errors.go

```diff
@@ -26,6 +26,17 @@ func (e *genericValidationError) Error() string {
 	return fmt.Sprintf(e.message, e.cause, formatLabelSet(e.series))
 }
 
+// labelValueTooLongError is a customized ValidationError, in that the cause and the series are
+// are formatted in different order in Error.
+type labelValueTooLongError struct {
+	labelValue string
+	series     []cortexpb.LabelAdapter
+}
+
+func (e *labelValueTooLongError) Error() string {
+	return fmt.Sprintf("label value too long for metric: %.200q label value: %.200q", formatLabelSet(e.series), e.labelValue)
+}
```

Comment:
[nit] Please move this below, to keep it just above its usage in `newLabelValueTooLongError()`

Reply:
Anyway, it's not a big deal. We can live with that. I will merge once CI passes.